plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/FeatureDevSessionContext.kt

```diff
@@ -97,17 +106,30 @@ class FeatureDevSessionContext(val project: Project, val maxProjectSizeBytes: Lo
         deferredResults.any { it.await() }
     }
 
+    private suspend fun ignoreFileByExtension(file: File, scope: CoroutineScope): Boolean = with(scope) {
+        return !isFileExtensionAllowed(file)
+    }
+
+    suspend fun ignoreFileByExtension(file: VirtualFile, scope: CoroutineScope): Boolean = ignoreFileByExtension(File(file.path), scope)
     suspend fun ignoreFile(file: VirtualFile, scope: CoroutineScope): Boolean = ignoreFile(File(file.path), scope)
 
     suspend fun zipFiles(projectRoot: VirtualFile): File = withContext(getCoroutineBgContext()) {
         val files = mutableListOf<VirtualFile>()
+        val ignoredExtensionMap = mutableMapOf<String, Int>()
         var totalSize: Long = 0
 
         VfsUtil.visitChildrenRecursively(
             projectRoot,
             object : VirtualFileVisitor<Unit>() {
                 override fun visitFile(file: VirtualFile): Boolean {
-                    val isFileIgnored = runBlocking { ignoreFile(file, this) }
+                    val isFileIgnoredByPattern = runBlocking { ignoreFile(file, this) }
+                    val isFileIgnoredByExtension = runBlocking { ignoreFileByExtension(file, this) }
+                    val isFileIgnored = isFileIgnoredByExtension || isFileIgnoredByPattern
+
+                    if (isFileIgnoredByExtension) {
+                        ignoredExtensionMap[file.extension.toString()] = (ignoredExtensionMap[file.extension] ?: 0) + 1
```

Comment:
```kt
val ignoredExtensionMap = mutableMapOf<String, Int>().withDefault { 0 }
...
ignoredExtensionMap[file.extension] += 1
```

Reply:
Since `file.extension` is nullable
```
public @Nullable @NlsSafe String getExtension()
```

This will cause unresolved reference